File: core/repository/orchestration_reasoning_engine.py
from __future__ import annotations

from typing import Dict, List


def reason_orchestration(services: List[str], dependencies: List[str]) -> Dict[str, object]:
    svc = sorted(set(str(s) for s in (services or []) if s))
    deps = sorted(set(str(d) for d in (dependencies or []) if d))
    edges = [{"from": svc[i], "to": svc[i + 1]} for i in range(len(svc) - 1)]
    return {"services": svc, "dependencies": deps, "flow_edges": edges}

File: src/repository/orchestrationReasoningEngine.ts
/**
 * Converted from Python: core/repository/orchestration_reasoning_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function reasonOrchestration(services: any, dependencies: any): any {
  var svc: any = py.sorted(py.toSet(py.iter(py.or2(services, () => ([]))).filter((s: any) => py.truthy(s)).map((s: any) => py.toStr(s))));
  var deps: any = py.sorted(py.toSet(py.iter(py.or2(dependencies, () => ([]))).filter((d: any) => py.truthy(d)).map((d: any) => py.toStr(d))));
  var edges: any = py.range(py.sub(py.len(svc), 1)).map((i: any) => ({"from": py.at(svc, i), "to": py.at(svc, py.add(i, 1))}));
  return {"services": svc, "dependencies": deps, "flow_edges": edges};
}
